File: ios/MeasureSize.m
//
//  MeasureSize.m
//  SqlPlay
//
//  Created by Shivam Kumar on 05/04/23.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(MeasureSize, NSObject)

RCT_EXTERN_METHOD(measureSize: (NSString *)text font:(NSString *)font size:(NSInteger)size resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/MeasureSize.swift
import Foundation
import UIKit
import React

@objc(MeasureSize)
class MeasureSize: NSObject, RCTBridgeModule {
    
    static func moduleName() -> String! {
        return "MeasureSize"
    }
    
    static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    @objc
    func measureSize(_ text: String, font: String, size: Int, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
        let font = UIFont(name: font, size: CGFloat(size)) ?? UIFont.systemFont(ofSize: CGFloat(size))
        let width = (text as NSString).size(withAttributes: [.font: font]).width
        resolve(width)
    }
}
